agent/installer/internal/algo/mock_ubuntu_with_error.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package algo
+
+type MockUbuntuWithError struct {
+	BaseK8sInstaller
+}
+
+func getEmptyStep(desc string, bki *BaseK8sInstaller) Step {
```

Comment:
Create your own step that counts the calls to do and undo. Assert these counters in the test.